### ISSUE TITLE
RoleGrant doesn't specify role type when DatabaseRole is used

### ISSUE THREAD

**Issue by dnabb** [Titan-Systems/titan#164]
The following code:

```python
db = Database(name='whatever')
parent_role = DatabaseRole(name='whatever_role', database=db)
child_role = DatabaseRole(name='whatever_subrole', database=db)
rg = RoleGrant(role=child_role, to_role=parent_role)

bp = Blueprint(
    resources=[db, parent_role, child_role, rg],
    )
bp.apply(session, plan=plan)
```

Produces the following SQL on bp.apply():

```sql
USE SECONDARY ROLES ALL
USE ROLE SYSADMIN
CREATE DATABASE WHATEVER DATA_RETENTION_TIME_IN_DAYS = 1 MAX_DATA_EXTENSION_TIME_IN_DAYS = 14
USE ROLE SYSADMIN
CREATE DATABASE ROLE WHATEVER.WHATEVER_ROLE
GRANT OWNERSHIP ON DATABASE ROLE WHATEVER.WHATEVER_ROLE TO ROLE USERADMIN COPY CURRENT GRANTS
USE ROLE SYSADMIN
CREATE DATABASE ROLE WHATEVER.WHATEVER_SUBROLE
GRANT OWNERSHIP ON DATABASE ROLE WHATEVER.WHATEVER_SUBROLE TO ROLE USERADMIN COPY CURRENT GRANTS
USE ROLE SECURITYADMIN
GRANT ROLE WHATEVER.WHATEVER_SUBROLE TO ROLE WHATEVER.WHATEVER_ROLE << This will fail
```

The proper syntax when granting database roles to each other is:

```sql
GRANT DATABASE ROLE WHATEVER.WHATEVER_SUBROLE TO DATABASE ROLE WHATEVER.WHATEVER_ROLE
```

Not specifying "DATABASE ROLE" everywhere will fail (with either syntax error or object does not exist.

**Comment by toadies** [comment]
Is there a YAML equivalent to above?

**Comment by teej** [comment]
Working on this issue made me realize there are a number of bugs with database roles. I am working on a fix but it requires a few changes to grant and role internals.

**Comment by andersbergren** [comment]
Ah, wish I'd seen this issue last week. Would reworking my RBAC template to uniquely name my database roles work? Or should I swap to using uniquely named account roles until database roles are fixed?

**Comment by teej** [comment]
I'll pick this back up over the weekend and roll it into v0.11 for next week

**Comment by teej** [comment]
Fixed in [v0.11.0](https://github.com/Titan-Systems/titan/releases/tag/v0.11.0). Use the new `DatabaseRoleGrant` resource.
